relay-server/src/actors/healthcheck.rs

```diff
@@ -88,12 +89,23 @@ impl Handler<IsHealthy> for Healthcheck {
             IsHealthy::Liveness => Box::new(future::ok(true)),
             IsHealthy::Readiness => {
                 if self.is_shutting_down {
-                    Box::new(future::ok(false))
-                } else if self.config.requires_auth() {
+                    return Box::new(future::ok(false));
+                }
+
+                let is_aggregator_full = Aggregator::from_registry()
+                    .send(AcceptsMetrics)
+                    .map_err(|_| ());
+                let is_authenticated: Self::Result = if self.config.requires_auth() {
                     Box::new(upstream.send(IsAuthenticated).map_err(|_| ()))
                 } else {
                     Box::new(future::ok(true))
-                }
+                };
+
+                Box::new(
+                    is_aggregator_full
+                        .join(is_authenticated)
+                        .map(|(a, b)| a && b),
+                )
```

Comment:
Does this mean that in production, we restart a Relay instance when the aggregator is full? If so, this could lead to the undesired behavior that we keep passing on metrics to the upstream because the aggregator gets respawned repeatedly, force-flushing its contents on every restart. IMO the correct behavior of a "full" aggregator is to keep rejecting new metrics until enough buckets have been flushed.

Reply:
no, the readiness check is afaik only used to determine whether a relay should receive traffic. the liveness check is used to determine whether relay should be shot down.

when a relay is not authenticated, we don't shoot it down, we wait until it is so it can receive traffic.

Reply:
OK, in that case we should be good!